airflow/providers/google/cloud/operators/datafusion.py

```diff
@@ -34,16 +35,25 @@
 from airflow.providers.google.cloud.operators.cloud_base import GoogleCloudBaseOperator
 from airflow.providers.google.cloud.triggers.datafusion import DataFusionStartPipelineTrigger
 from airflow.providers.google.cloud.utils.datafusion import DataFusionPipelineType
+from airflow.providers.google.cloud.utils.helpers import resource_path_to_dict
 
 if TYPE_CHECKING:
     from airflow.utils.context import Context
 
 
 class DataFusionPipelineLinkHelper:
-    """Helper class for Pipeline links."""
+    """
+    Helper class for Pipeline links.
+
+    .. warning::
+        This class is deprecated. Consider using ``resource_path_to_dict()`` instead.
+    """
 
     @staticmethod
     def get_project_id(instance):
+        warnings.warn(
+            "DataFusionPipelineLinkHelper is deprecated. Consider using resource_path_to_dict() instead."
```

Comment:
You need to raise `AirflowProviderDeprecationWarning`

Reply:
Thanks for the comment. Fixed. Please re-review.